### PR TITLE
Add command line parameter that turns on leak tracking.

### PR DESCRIPTION
**Pull request by polina-c**
CUJs that should be enabled:

1. Manual run of tests with CLI
2. Manual run of tests with IDE
3. https://github.com/flutter/flutter/issues/138742
4. https://github.com/flutter/flutter/issues/138741

### CODE REVIEW

File: packages/flutter/test/flutter_test_config.dart [hunk 1]
```diff
@@ -11,6 +11,17 @@ import 'package:leak_tracker/leak_tracker.dart';
 import '_goldens_io.dart'
   if (dart.library.html) '_goldens_web.dart' as flutter_goldens;
 
+/// If true, leak tracking will be enabled for all tests `testWidgetsWithLeakTracking`.
+///
+/// By default, the constant is false.
+/// To enable the leak tracking for all tests, either pass the compilation flag
+/// `--dart-define=flutter_test_config.leak_tracking=true` or
+/// temporarily pass `defaultValue = true` to `fromEnvironment` in the constant definition.
```
Comment (jacob314) [inline packages/flutter/test/flutter_test_config.dart:19]: nit: remove the line starting with 

"or temporarily pass `defaultValue = true` to `fromEnvironment` in the constant definition.
I would think passing in the `--dart-define` should be sufficient for all users.

**Review by jacob314 — COMMENTED** [review]
lgtm

**Comment by polina-c** [comment]
@godofredoc, can you review to make sure it is a right way to parameterize tests, to enable (1) use cases we discussed on Friday and (2) users to run with and without leak tracking?

**Comment by polina-c** [comment]
@keyonghan

File: packages/flutter/test/flutter_test_config.dart [hunk 2]
```diff
@@ -11,6 +11,17 @@ import 'package:leak_tracker_flutter_testing/leak_tracker_flutter_testing.dart';
...
+///
+/// To enable leak tracking for an individual test file, add the line to the test `main`:
+/// `LeakTesting.settings = LeakTesting.settings.withTrackedAll()`.
+const bool _kLeakTracking = bool.fromEnvironment('flutter_test_config.leak_tracking');
```
Comment (keyonghan) [inline packages/flutter/test/flutter_test_config.dart:23]:
```suggestion
const bool _kLeakTracking = bool.fromEnvironment('LEAK_TRACKING');
```

**Review by keyonghan — APPROVED** [review]
The `env` variable name LGTM.

**Pull request merged**